.github/workflows/pkg.yml

```diff
@@ -33,15 +42,72 @@ jobs:
 
           # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
           # skip-build-cache: true
+
+      - name: Verify lint report artifact
+        if: always()
+        run: test -f golangci-lint-report.xml && cat golangci-lint-report.xml || true
+
+      - name: Upload lint report artifact
+        if: always()
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+        with:
+          name: golangci-lint-report
+          path: golangci-lint-report.xml
+
   build-test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
         with:
           go-version-file: 'go.mod'
-      - name: build
+
+      - name: Build
         run: go build -v ./...
-      - name: test
-        run: go test -count=1 -race ./...
-        
+
+      - name: Unit Tests
+        run: go test -race ./... -coverpkg=./... -coverprofile=coverage.txt $1 | tee $OUTPUT_FILE
```

Comment:
@chudilka1 this is not working properly. Failing tests and detected races no longer fail the workflow. For example: https://github.com/smartcontractkit/chainlink-relay/actions/runs/5283496635/jobs/9559819542